### ISSUE TITLE
[MOBILE DESIGN]: GIBCT EYB - Responsive tables SHOULD have HTML updated for better mobile presentation

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#14420]
# [508-defect-3](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-3)

**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Point of Contact

**VFS Point of Contact:** _Trevor_

## User Story or Problem Statement

As a mobile user, I want to see a header for the responsive table rows, so I understand what each data point means.

## Details

The responsive table that's deployed on the GIBCT needs a couple of modifications to be fully inline with the recommended version that will become a design system component. Specifically:

1. The `tbody > tr > th` needs a `role="rowheader"` attribute
1. The `tbody > tr > th` needs a `<dfn>` added that is hidden at desktop size, and shown at mobile size

## Acceptance Criteria

- [ ] Code is updated to match the HTML snippet and [comment on 13825](https://github.com/department-of-veterans-affairs/va.gov-team/issues/13825#issuecomment-700810625) from VSA specialist
- [ ] DFN labels are appearing at mobile sizes
- [ ] HTML validates in the W3C validator
- [ ] Zero axe-core violations are shown

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Environment

* https://staging.va.gov/gi-bill-comparison-tool/profile/11006124 or any other profile view that has the responsive table

## Solution (if known)

```diff
<tbody>
    <tr role="row">
-     <th class="school-name-cell" scope="row" role="cell" tabindex="-1">...</th>
+     <th class="school-name-cell" role="rowheader" scope="row" tabindex="-1"> <!-- add rowheader role; set scope of th to row -->
+        <dfn>SMC letter designation</dfn> <!-- provides the column heading for mobile, but hides in desktop view -->
+        <span>SMC-K</span> <!-- the value of the row header -->
     </th>
     <td role="cell"><dfn>Monthly payment (in&nbsp;U.S.&nbsp;$)</dfn><span class="currency">110.31</span></td>
     <td role="cell"><dfn>How this payment variation works</dfn><span class="paragraph">If you qualify for SMC-K, we add this rate to your basic disability compensation rate for any disability rating from 0% to 100%. We also add this rate to all SMC basic rates except SMC-O, SMC-Q, and SMC-R. You may receive 1 to 3 SMC-K awards in addition to basic and SMC rates.</span></td>
  </tr>
```

## Screenshots or Trace Logs

![Screen Shot 2020-10-05 at 3 54 30 PM](https://user-images.githubusercontent.com/934879/95131939-91b97680-0724-11eb-8bbd-29ec6c5ad24e.png)

**Comment by zurbergram** [comment]
![Screen Shot 2020-10-08 at 11.56.17 AM.png](https://images.zenhubusercontent.com/5bfd74885cb43a45a18acb1b/c42715ad-10b4-4064-93c3-44285467b7c9)

**Comment by zurbergram** [comment]
A cleaner solution?

![Screen Shot 2020-10-08 at 12.03.58 PM.png](https://images.zenhubusercontent.com/5bfd74885cb43a45a18acb1b/ca701069-aac8-4b89-830b-133957060a6f)

**Comment by zurbergram** [comment]
Fixed extension row styling and changed mobile header dfn
![image.png](https://images.zenhubusercontent.com/5bfd74885cb43a45a18acb1b/41aec00d-c188-4b96-bc75-d0b56f44e467)

**Comment by 1Copenut** [comment]
@emilywaggoner && @CrystabelReiter FYSA I met with the BAH team earlier today to discuss this responsive table for the Estimate Your Benefits work. The team is working on a way to differentiate extension campuses from the main campus on mobile views, and is working through the final visual polish.

**Comment by elder-joseph** [comment]
Associated PR: https://github.com/department-of-veterans-affairs/vets-website/pull/14506

**Comment by elder-joseph** [comment]
This issue is implemented as described, but styling changes were requested. This is being put on hold until the styling changes are documented.

**Comment by watersemma** [comment]
Here is the proposed mobile table updates. https://bahdigital.invisionapp.com/share/B8IAF9K75NU Screenshot attached for your convenience. ![Screen Shot 2020-10-21 at 2.31.48 PM.png](https://images.zenhubusercontent.com/5f7e1db6d6b9a95b02babbae/9f7b821e-0d00-4902-9e14-51c65da93084)

**Comment by delli-gatti-michael** [comment]
Local testing complete

![Screen Shot 2020-10-26 at 3.03.12 PM.png](https://images.zenhubusercontent.com/5c07f000391a6f6357adc430/2eeb0f1e-3a17-4974-8fe8-ddd92a8bf722)

**Comment by delli-gatti-michael** [comment]
@department-of-veterans-affairs/uxbah  @1Copenut 
This is on staging and ready for review

**Comment by JenLJ** [comment]
UX Review 1/2:  This checks out for me!

**Comment by elder-joseph** [comment]
CSS Fix PR: https://github.com/department-of-veterans-affairs/vets-website/pull/14739

**Comment by delli-gatti-michael** [comment]
The second PR is now on Staging.

**Comment by watersemma** [comment]
UX Review 2: Pending 30 px margin on either side. 
![Mock.png](https://images.zenhubusercontent.com/5f7e1db6d6b9a95b02babbae/54e9df60-0ef6-40f8-a8db-1cbd7d481952)![Staging.png](https://images.zenhubusercontent.com/5f7e1db6d6b9a95b02babbae/f44efc21-c093-49a4-b482-df132b27812c)
![Mock 2.png](https://images.zenhubusercontent.com/5f7e1db6d6b9a95b02babbae/d3bcfd24-da4a-4856-ad9b-b59142b8447c)

**Comment by delli-gatti-michael** [comment]
PR \#3:
https://github.com/department-of-veterans-affairs/vets-website/pull/14763

**Comment by amyknox** [comment]
UX Review 1/2: Looks good to me.
@watersemma - could you review & weigh in?

**Comment by watersemma** [comment]
UX Review 2/2: Looks good. One comment, can we adjust the text alignment here for location? 
\![14420.png](https://images.zenhubusercontent.com/5f7e1db6d6b9a95b02babbae/8149925c-50bc-473c-ae82-1890ee72e359)

**Comment by amyknox** [comment]
Hey @watersemma - could you confirm if you're signing off on this story or the alignment still needs to be adjusted?  Thanks!

**Comment by watersemma** [comment]
@amyknox sorry for the confusion, alignment needs adjustment.

**Comment by delli-gatti-michael** [comment]
PR \#4 is now on Staging:
https://github.com/department-of-veterans-affairs/vets-website/pull/14782

**Comment by watersemma** [comment]
UX Review 1/2: This checks out for me!

**Comment by JenLJ** [comment]
UX Review 2/2:  This also checks out for me!

**Comment by sonntag-adam** [comment]
@1Copenut this is ready for review in staging.

**Comment by 1Copenut** [comment]
@sonntag-adam I spot checked this in staging with iOS VoiceOver on an iPhone 8. The table is working really well visually and how it announces the information. Moving to close.
